src/lists/list.jsx

```diff
@@ -92,35 +95,32 @@ const List = React.createClass({
       ...other,
     } = this.props;
 
+    let hasSubheader = false;
+
+    if (subheader) {
+      hasSubheader = true;
+    } else {
+      const firstChild = React.Children.toArray(children)[0];
+      if (React.isValidElement(firstChild) && firstChild.type === Subheader) {
```

Comment:
@oliviertassinari I'm gonna make an issue for this kind of things so we can fully discuss what we should do :+1: Thanks for pointing it out